AndroidCompat/src/main/java/xyz/nulldev/androidcompat/io/sharedprefs/JavaSharedPreferences.kt

```diff
@@ -0,0 +1,168 @@
+package xyz.nulldev.androidcompat.io.sharedprefs
+
+import android.content.SharedPreferences
+import com.russhwolf.settings.ExperimentalSettingsApi
+import com.russhwolf.settings.ExperimentalSettingsImplementation
+import com.russhwolf.settings.JvmPreferencesSettings
+import com.russhwolf.settings.serialization.decodeValue
+import com.russhwolf.settings.serialization.encodeValue
+import com.russhwolf.settings.set
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.builtins.SetSerializer
+import kotlinx.serialization.builtins.nullable
+import kotlinx.serialization.builtins.serializer
+import java.util.prefs.PreferenceChangeListener
+import java.util.prefs.Preferences
+
+@OptIn(ExperimentalSettingsImplementation::class, ExperimentalSerializationApi::class, ExperimentalSettingsApi::class)
+class JavaSharedPreferences(key: String) : SharedPreferences {
+    private val javaPreferences = Preferences.userRoot().node("suwayomi/tachidesk/$key")
+    private val preferences = JvmPreferencesSettings(javaPreferences)
+    private val listeners = mutableMapOf<SharedPreferences.OnSharedPreferenceChangeListener, PreferenceChangeListener>()
+
+    // TODO: 2021-05-29 Need to find a way to get this working with all pref types
```

Comment:
now or later?

Reply:
I would say later, I dont see it being used at all